scripts/sanity_chk/sanitylib.py

```diff
@@ -1602,6 +1602,11 @@ def check_build_or_run(self, build_only=False, enable_slow=False, device_testing
 
         self.build_only = not (not _build_only and runnable)
         self.run = not self.build_only
+
+        # we asked for build-only on the command line
+        if opt_build_only:
+            self.build_only = True
```

Comment:
I think that when this check is just after `self.run = not self.build_only` we could end up (at least in theory) with both `self.build_only` and `self.run` == `True`. Isn't this an issue?

Reply:
I think I got it now. This is actually fine and would happen if the test is perfectly able to be run on a platform (and doesn't have build_only set to true) but the user used option from the cli to only build. I am right now?

Reply:
Correct.  This trying to cleanup the use of build_only in several places having slightly different meanings.